src/helpers/format/format.ts

```diff
@@ -130,9 +145,46 @@ function applyTextInternalFormat(
       case "STRING":
         formattedValue += token.value;
         break;
+      case "REPEATED_CHAR":
+        formattedValue += REPEATED_CHAR_PLACEHOLDER + token.value;
+        break;
     }
   }
-  return formattedValue;
+  return repeatCharToFitWidth(formattedValue, formatWidth);
+}
+
+function repeatCharToFitWidth(formattedValue: string, formatWidth?: FormatWidth): string {
+  const placeholderIndex = formattedValue.indexOf(REPEATED_CHAR_PLACEHOLDER);
+  if (placeholderIndex === -1) {
+    return formattedValue;
+  }
+  const prefix = formattedValue.slice(0, placeholderIndex);
+  const suffix = formattedValue.slice(placeholderIndex + REPEATED_CHAR_PLACEHOLDER.length + 1);
+  const repeatedChar = formattedValue[placeholderIndex + REPEATED_CHAR_PLACEHOLDER.length];
+
+  function getTimesToRepeat() {
+    if (!formatWidth) {
+      return { timesToRepeat: 0, padding: "" };
+    }
+    const widthTaken = formatWidth.measureText(prefix + suffix);
+    const charWidth = formatWidth.measureText(repeatedChar);
+    const availableWidth = formatWidth.availableWidth - widthTaken;
+    if (availableWidth <= 0) {
+      return { timesToRepeat: 0, padding: "" };
+    }
+
+    const timesToRepeat = Math.floor(availableWidth / charWidth);
+
+    const remainingWidth = availableWidth - timesToRepeat * charWidth;
+    const paddingChar = "\u2009"; // thin space
+    const paddingWidth = formatWidth.measureText(paddingChar);
+    const padding = paddingChar.repeat(Math.floor(remainingWidth / paddingWidth));
+
+    return { timesToRepeat, padding };
+  }
+
+  const { timesToRepeat, padding } = getTimesToRepeat();
+  return prefix + repeatedChar.repeat(timesToRepeat) + padding + suffix;
```

Comment:
padding should be before the repeated chars

Reply:
Why ? For a format like `*a0.00` it make more sense to have the padding after the repeated character. But there is probably an example where the padding make more sense before the repeated character.

I don't mind having the padding before, but both are equally problematic no ? Or do you see a specific reason it's better before ?

Reply:
> Or do you see a specific reason it's better before ?

not really, except mimicking what Gsheet does

Reply:
aaaaaand Excel does add the padding after the repeated character 😅 I vote to leave it as it is 🤷‍♂️

Reply:
:+1: